### PR TITLE
Improve the Warnings about missing clipboard registers

### PR DESCRIPTION
**Pull request by chrisbra**
- Make it translatable
- Use a different warning, when clipboard support was not compiled in

### CODE REVIEW

File: src/message.c [hunk 1]
```diff
@@ -4071,7 +4070,11 @@ msg_warn_missing_clipboard(void)
 {
     if (!global_busy && !did_warn_clipboard)
     {
-       msg(_(warn_clipboard));
+#ifdef FEAT_CLIPBOARD
+       msg(_("W23: Clipboard register not available, using register 0"));
+#else
+       msg(_("W24: Clipboard register not available"));
```
Comment (saccarosium) [inline src/message.c]: I think this should be a little more specific about the fact that vim doesn't have this functionality compiled with ideally a link to a doc detailing what is `+clipboard`. Mainly for new users that they will likely now about `+clipboard` and the fact that you need to recompile vim.

Probably something like: "Clipboard register is not available in this build of vim. To learn more read `:h clipboard`"

**Review by saccarosium — CHANGES_REQUESTED** [review]

**Comment by vim-ml** [comment]
On 2024-11-16, Luca Saccarola wrote:
> @saccarosium requested changes on this pull request.
> 
> -------------------------------------------------------------------------------
> 
> In src/message.c:
> 
> > @@ -4071,7 +4070,11 @@ msg_warn_missing_clipboard(void)
>  {
>      if (!global_busy && !did_warn_clipboard)
>      {
> -       msg(_(warn_clipboard));
> +#ifdef FEAT_CLIPBOARD
> +       msg(_("W23: Clipboard register not available, using register 0"));
> +#else
> +       msg(_("W24: Clipboard register not available"));
> 
> I think this should be a little more specific about the fact that vim doesn't
> have this functionality compiled with ideally a link to a doc detailing what is
> +clipboard. Mainly for new users that they will likely now about +clipboard and
> the fact that you need to recompile vim.
> 
> Probably something like: "Clipboard register is not available in this build of
> vim. To learn more read :h clipboard"

But new users should know about, or be made aware of, ":help W23".
I don't think we should be turning error messages into essays when
we have a very good help system.

Regards,
Gary

**Comment by saccarosium** [comment]
TLDR: at least try to tell the user why the clipboard is unavailable.

> But new users should know about, or be made aware of, ":help W23".

IMHO Errors should be helpful and guide you to the solution. The all `W*` is not really user friendly TBH (but that is another discussion). If I'm a new user how I suppose to know that that `W24` is a helptag to use in the docs? I've used vim for like 6 years and discovered this not even that long ago and I'm certain that will confuse people at least from my personal experience teaching vim and helping others triage issues.

> I don't think we should be turning error messages into essays

Man you read really short essays :)

**Comment by saccarosium** [comment]
LGTM!
